cmd/orch-tester/orch_tester.go

```diff
@@ -0,0 +1,558 @@
+	}
+
+	body, err := ioutil.ReadAll(res.Body)
+	defer res.Body.Close()
+	if err != nil {
+		return "", err
+	}
+
+	if res.StatusCode < 200 || res.StatusCode >= 300 {
+		return "", errors.New(string(body))
+	}
+
+	var startStreamsRes streamerModel.StartStreamsRes
+	if err := json.Unmarshal(body, &startStreamsRes); err != nil {
+		return "", err
+	}
+
+	return startStreamsRes.BaseManifestID, nil
+}
+
+func (s *streamerClient) getFinishedStats(mid string) (*streamerModel.Stats, error) {
+	var stats streamerModel.Stats
+
+	for !stats.Finished {
+
+		time.Sleep(5 * time.Second)
+
+		req, err := http.NewRequest("GET", s.streamer+"/stats?base_manifest_id="+url.QueryEscape(mid), nil)
+		if err != nil {
+			return nil, err
+		}
+
+		res, err := s.client.Do(req)
+		if err != nil {
+			return nil, err
+		}
+
+		body, err := ioutil.ReadAll(res.Body)
+		defer res.Body.Close()
+		if err != nil {
+			return nil, err
+		}
+
+		if res.StatusCode < 200 || res.StatusCode >= 300 {
+			return nil, errors.New(string(body))
+		}
+
+		if err := json.Unmarshal(body, &stats); err != nil {
+			return nil, err
+		}
+	}
+
+	return &stats, nil
+}
+
+func (s *streamerClient) avgSegDuration() (float64, error) {
+	val, err := s.queryVectorMetric(
+		"rate(livepeer_source_segment_duration_seconds_sum[1m])/rate(livepeer_source_segment_duration_seconds_count[1m])",
+	)
+	if err != nil {
+		return 0, err
+	}
+	valFloat := float64(0)
+	if val.Len() > 0 {
+		valFloat = float64((*val)[0].Value)
+		if math.IsNaN(valFloat) {
+			return 0, nil
+		}
+	}
+	return valFloat, nil
+}
+
+func (s *streamerClient) avgUploadTime() (float64, error) {
+	val, err := s.queryVectorMetric(
+		"rate(livepeer_upload_time_seconds_sum[1m])/rate(livepeer_upload_time_seconds_count[1m])",
+	)
+	if err != nil {
+		return 0, err
+	}
+	valFloat := float64(0)
+	if val.Len() > 0 {
+		valFloat = float64((*val)[0].Value)
+		if math.IsNaN(valFloat) {
+			return 0, nil
+		}
+	}
+	return valFloat, nil
+}
+
+func (s *streamerClient) avgDownloadTime() (float64, error) {
+	val, err := s.queryVectorMetric(
+		"rate(livepeer_download_time_seconds_sum[1m])/rate(livepeer_download_time_seconds_count[1m])",
+	)
+	if err != nil {
+		return 0, err
+	}
+	valFloat := float64(0)
+	if val.Len() > 0 {
+		valFloat = float64((*val)[0].Value)
+		if math.IsNaN(valFloat) {
+			return 0, nil
+		}
+	}
+	return valFloat, nil
+}
+
+func (s *streamerClient) avgRoundTripTime() (float64, error) {
+	val, err := s.queryVectorMetric(
+		"rate(livepeer_transcode_overall_latency_seconds_sum[1m])/rate(livepeer_transcode_overall_latency_seconds_count[1m])",
+	)
+	if err != nil {
+		return 0, err
+	}
+	valFloat := float64(0)
+	if val.Len() > 0 {
+		valFloat = float64((*val)[0].Value)
+		if math.IsNaN(valFloat) {
+			return 0, nil
+		}
+	}
+	return valFloat, nil
+}
+
+func (s *streamerClient) queryErrorCounts() ([]apiModels.Error, error) {
+	errors := make(map[string]int)
+	ctx, cancel := context.WithTimeout(context.Background(), 8*time.Second)
```

Comment:
Can we use the top level defined `httpTimeout` constant? Same comment applies for other context with timeout creation code.

Reply:
fixed in 6d37cae